### ISSUE TITLE
incorrect resources configuration for grid jobs

### ISSUE THREAD

**Issue by haqle314** [marbl/canu#1166]
I'm running canu 1.8 with slurm. When nodes with the most memory do not also have the highest core count, `canu` requests an invalid core/mem combination. This is the relevant part from `canu` output

    -- Detected Slurm with 'sinfo' binary in /usr/bin/sinfo.
    -- Detected Slurm with 'MaxArraySize' limited to 24999 jobs.
    -- 
    -- Found   1 host  with  20 cores and   94 GB memory under Slurm control.
    -- Found  17 hosts with  24 cores and  125 GB memory under Slurm control.
    -- Found   1 host  with  32 cores and  125 GB memory under Slurm control.
    -- Found   7 hosts with  16 cores and   54 GB memory under Slurm control.
    -- Found   1 host  with  16 cores and   62 GB memory under Slurm control.
    -- Found   1 host  with  16 cores and   54 GB memory under Slurm control.
    -- Found   1 host  with  24 cores and  125 GB memory under Slurm control.
    -- Found   2 hosts with  24 cores and  125 GB memory under Slurm control.
    -- Found   2 hosts with  32 cores and  125 GB memory under Slurm control.
    -- Found  14 hosts with  16 cores and   62 GB memory under Slurm control.
    -- Found   2 hosts with  16 cores and   62 GB memory under Slurm control.
    -- Found   1 host  with  20 cores and  377 GB memory under Slurm control.
    -- Found   8 hosts with  16 cores and   62 GB memory under Slurm control.
    -- Found   2 hosts with  24 cores and  503 GB memory under Slurm control.
    -- Found   1 host  with  32 cores and   94 GB memory under Slurm control.
    --
    --                     (tag)Threads
    --            (tag)Memory         |
    --        (tag)         |         |  algorithm
    --        -------  ------  --------  -----------------------------
    -- Grid:  meryl     27 GB    8 CPUs  (k-mer counting)
    -- Grid:  hap       16 GB   16 CPUs  (read-to-haplotype assignment)
    -- Grid:  cormhap   23 GB    8 CPUs  (overlap detection with mhap)
    -- Grid:  obtovl    24 GB    8 CPUs  (overlap detection)
    -- Grid:  utgovl    24 GB    8 CPUs  (overlap detection)
    -- Grid:  ovb        4 GB    1 CPU   (overlap store bucketizer)
    -- Grid:  ovs       32 GB    1 CPU   (overlap store sorting)
    -- Grid:  red       11 GB    4 CPUs  (read error detection)
    -- Grid:  oea        8 GB    1 CPU   (overlap error adjustment)
    -- Grid:  bat      503 GB   32 CPUs  (contig construction with bogart)
    -- Grid:  gfa       32 GB   32 CPUs  (GFA alignment and processing)

As you can see, nodes with 503 GB mem only has 24 cores, nodes with 32 cores has less mem. However, `bat` step is configured to use both 32 CPUs and 503 GB of memory.

**Comment by brianwalenz** [comment]
Interesting.  What was your canu command?

Use batMemory and batThreads to explicitly limit it to a valid configuration - batMemory=100 batThreads=16 is probably sufficient.

**Comment by haqle314** [comment]
Nothing special, I just needed to specify some arbitrary long time. Without requesting runtime, the grid is configured to kill jobs after 1 hour.

    canu \
        -p prefix \
        -d out/foo \
        genomeSize=2.4g \
        -gridOptions="-t 3-0" \
        -pacbio-raw data/fastq/*.fastq.gz

Yes, I figured that I can limit CPU/mem for individual jobs. What I'm doing right now is start canu, cancel the job after the resources allocation step. Then I rerun canu with extra limits like `-batMemory=125`. But it'd be nice if I don't have to add this extra step.
